### PR TITLE
[MM-60676] Fix incorrect callback URL in setup flow

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost-plugin-github/pull/811 introduced a regression: The callback URL shown as part of the setup flow was missing the SiteURl.

This PR fixes the issue. On top of that, it unifies the handling of the SiteURL using two helper methods. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60676

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:  I like the helper funcs.

**Comment by hanzei** [comment]
@Kshitij-Katiyar @raghavaggarwal2308 Gentle reminder to review this fix for a regression

File: server/plugin/api.go [hunk 1]
```diff
@@ -817,9 +818,7 @@ func (p *Plugin) searchIssues(c *UserContext, w http.ResponseWriter, r *http.Req
 }
 
 func (p *Plugin) getPermaLink(postID string) string {
-	siteURL := *p.client.Configuration.GetConfig().ServiceSettings.SiteURL
-
-	return fmt.Sprintf("%v/_redirect/pl/%v", siteURL, postID)
+	return getSiteURL(p.client) + "/" + path.Join("_redirect", "pl", postID)
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go:821]:
```suggestion
	return fmt.Sprintf("%s/%s", getSiteURL(p.client), path.Join("_redirect", "pl", postID))
```

File: server/plugin/plugin.go [hunk 2]
```diff
@@ -570,8 +570,7 @@ func (p *Plugin) getOAuthConfigForChimeraApp(scopes []string) *oauth2.Config {
 
 	authURL.Path = path.Join(authURL.Path, "oauth", "authorize")
 	tokenURL.Path = path.Join(tokenURL.Path, "oauth", "token")
-
-	redirectURL, _ := url.Parse(fmt.Sprintf("%s/plugins/github/oauth/complete", *p.client.Configuration.GetConfig().ServiceSettings.SiteURL))
+	redirectURL, _ := url.Parse(path.Join(getPluginURL(p.client), "oauth", "complete"))
```
Comment (raghavaggarwal2308) [inline server/plugin/plugin.go:573]: If it is returning an error, should we handle and log it here?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Comment by hanzei** [comment]
@raghavaggarwal2308 I've updated the PR to ues `url.JoinPath` and check its error.

**Review by raghavaggarwal2308 — APPROVED** [review]
@hanzei LGTM 👍🏽

**Comment by hanzei** [comment]
@AayushChaudhary0001 Gentle reminder to review the PR. I would like to fix the regression that is on `master`.

File: server/plugin/utils.go [hunk 3]
```diff
@@ -370,6 +372,29 @@ func isValidURL(rawURL string) error {
 	return nil
 }
 
+func buildPluginURL(client *pluginapi.Client, elem ...string) (string, error) {
+	siteURL, err := getSiteURL(client)
+	if err != nil {
+		return "", err
+	}
+
+	redirectURL, err := url.JoinPath(siteURL, append([]string{"foo", Manifest.Id}, elem...)...)
```
Comment (raghavaggarwal2308) [inline server/plugin/utils.go:381]: @hanzei What is "foo" here?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin/utils.go (hunk 3 above)
Comment (hanzei) [reply server/plugin/utils.go]: Great catch! I highly appreciate the double-checking. Fixed via ad80dd5.

**Review by AayushChaudhary0001 — APPROVED** [review]
Tested this PR for the following functionalities:-
- Basic OAuth flow.
- Connection and disconnection of user's account.

The PR was working fine for the following above conditions, LGTM. Approved.

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**